vars/publishNotification.groovy

```diff
@@ -1,18 +1,19 @@
 void call(Map args = [:]) {
+    def manifest = (args.credentialsId == 'BUILD_NOTICE_WEBHOOK') ? "Manifest: ${INPUT_MANIFEST}" : null
     text = ([
         "${args.icon} ${JOB_NAME} [${BUILD_NUMBER}] ${args.message}",
         "Build: ${BUILD_URL}",
-        "Manifest: ${INPUT_MANIFEST}",
+        manifest,
         args.extra
     ] - null).join("\n")
 
-    withCredentials([string(credentialsId: 'BUILD_NOTICE_WEBHOOK', variable: 'BUILD_NOTICE_WEBHOOK')]) {
+    withCredentials([string(credentialsId: args.credentialsId, variable: 'WEBHOOK_URL')]) {
         sh ([
             args.script ?: 'curl',
             '-XPOST',
             '--header "Content-Type: application/json"',
             "--data '{\"result_text\":\"${text}\"}'",
-            "\"${BUILD_NOTICE_WEBHOOK}\""
+            "\$" + 'WEBHOOK_URL'
         ].join(' '))
```

Comment:
Does this actually resolve $WEBHOOK_URL? Doesn’t look like it according to the test. Should just be $WEBHOOK_URL

Reply:
Yes. Tested the notifications

Reply:
The code as is is still not readable and the + unnecessary. You want the value of $WEBHOOK_URL so just write that?

Reply:
Made the change.